### DISCUSSION TITLE
ClientRequest.write_bytes taking too long

### DISCUSSION THREAD

**Discussion by matias-lg** [aio-libs/aiohttp#7176] [Q&A]
Currently I am sending some graphql queries using the typical post request:
```python
        async with aiohttp.ClientSession() as session:
            async with session.post(url = BASE_URL, json = {'query': document}, headers=headers) as response:
                 # do stuff with the response
```
This is executed by different fastAPI endpoints that get hit at the same time. The post requests manage to run in parallel but
every single one is taking a bit too long when comparing them with just querying `requests.post` (this takes ~1.5 seconds), with aiohttp they take around 4 seconds each. When profiling each call most of the time (2 seconds) are caused by `ClientRequest.write_bytes`, am I using the functions wrong? Is it just an expected overhead? Any help would be apreciated.

**Comment by webknjaz** [comment]
I think there's not enough context here. We'd need a more complete reproducer to judge what's happening in your case.

**Comment by Dreamsorcerer** [comment]
It's possible you're measuring the wrong thing, especially if you have multiple requests being handled at the same time.

requests is synchronous, so your entire application stops and waits for it to finish the write.
If you are using aiohttp and are simply logging the time when you make the request, and the time your function resumes after completing the request, then that is probably not the time it took to send the request. Your application probably jumped between several other tasks before returning to that function.

Test it with a simple script (without a web application) that sends a single request and see if you observe aiohttp taking twice as much time. If the difference is not there in a simple script, then I'm pretty sure it's just the problem of measuring it described above.
